### ISSUE TITLE
Show function arguments does not work for debugging w/ GDB

### ISSUE THREAD

**Issue by PokhodenkoSA** [numba/numba#7414]
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

Python code:

```python
from numba import njit

@njit(debug=True)
def sum(a, b):
    c = a + b  # break-here
    return c

print(sum(1, 2))
```

Debug session:

```bash
gdb -q --args python chk_info_args.py
(gdb) break chk_info_args.py:6
(gdb) run
...
Thread 1 "python" hit Breakpoint 1, __main__::sum$241 () at chk_info_args.py:6
6           c = a + b
(gdb) info args
No arguments.
```

It seems that `info args` does not show arguments.

**Comment by stuartarchibald** [comment]
Thanks for the report. I'm pretty sure the bugs are one or a combination of:

Args are not enumerated (most likely):
https://github.com/numba/numba/blob/4911192c5b51f3ecbb76a12b74372979378c43a1/numba/core/debuginfo.py#L141

Subroutine type lacks metadata in the type decl:
https://github.com/numba/numba/blob/4911192c5b51f3ecbb76a12b74372979378c43a1/numba/core/debuginfo.py#L241

**Comment by stuartarchibald** [comment]
https://github.com/numba/numba/pull/7177 fixes.

**Comment by stuartarchibald** [comment]
Example under #7177:
```python
$ cat issue7414.py
from numba import njit
import numpy as np

@njit(debug=True)
def sum(a, b):
    c = a + b  # break-here
    return c

sum(1, np.ones((5,)))

$ NUMBA_OPT=0 NUMBA_EXTEND_VARIABLE_LIFETIMES=1 gdb -q --args python issue7414.py
(gdb) break issue7414.py:6
(gdb) r
Breakpoint 1, __main__::sum$241 (a=1, b=...) at issue7414.py:6
6           c = a + b  # break-here
(gdb) info args
a = 1
b = {meminfo = 0x555556690cd0 "\002", parent = 0x7fffdd8e7bd0 "\004", nitems = 5, itemsize = 8, data = 0x5555562502a0, 
  shape = {5}, strides = {8}}
(gdb) info locals
c = {meminfo = 0x0, parent = 0x0, nitems = 0, itemsize = 0, data = 0x0, shape = {0}, strides = {0}}
```

**Comment by stuartarchibald** [comment]
Closed by #7177.
